src/droneSimulator.py

```diff
@@ -0,0 +1,83 @@
+import env2d
+import model2d
+import math
+
+
+def applyAccel(posPrev, velPrev, accel, timeStep):
+	vel = velPrev + accel * timeStep
+	pos = posPrev + velPrev * timeStep + 0.5 * accel * timeStep**2
+	return (pos, vel)
+
+def applyForce(force, drone, timeStep):
+	forceX = force[0]
+	forceY = force[1]
+	(newX, newVx) = applyAccel(drone.pos.x, drone.vel.vx, forceX / drone.mass, timeStep)
+	(newY, newVy) = applyAccel(drone.pos.y, drone.vel.vy, forceY / drone.mass, timeStep)
+
+	alpha = drone.getAngularAccel()
+	(newT, newW) = applyAccel(drone.pos.t, drone.vel.w, alpha, timeStep)
+	drone.pos = DronePos(newX, newY, newT)
+	drone.vel = DroneVel(newVx, newVy, newW)
+
+def doOneStep(env, drone, timeStep):
+	# The forces on the drone, currently, are gravity, the drone's thrust,
+	# and the resistance from the wind.
+	thrust = drone.getThrust()
+	forceX = env.gravityX() + thrust[0]
+	forceY = env.gravityY() + thrust[1]
+	applyForce([forceX, forceY], drone, timeStep)
+
+def runSimulation(env, drone, controller, timeStep, totalTime):
```

Comment:
The simulation ought to be separated from doOneStep/applyForce etc.  Or at least, the simulation should be pulled out into its own class and this main method/class ought to consume it.